### PR TITLE
feat: add registration form footer

### PR DESCRIPTION
**Pull request by bowenzhu1**
## Notion ticket link

[Registration form footer](https://www.notion.so/uwblueprintexecs/Registration-Form-Footer-c22a2125cce546238cd2014e0dd73827)

## Implementation description
* Currently using placeholder alerts for navigating outside of registration steps (initial screen + payments)
* Also using placeholder alert for incomplete field error 

## Steps to test
1. Navigating through steps in /register/camp/:id should work, expected alerts should pop up (incomplete form, payment, back to initial screen)

## What should reviewers focus on?
* Step navigation using index (stepNavigation function)

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

## Screenshots (for phone/tablet screen sizes)
Screen Shot 2022-11-26 at 2 25 18 PM
Screen Shot 2022-11-26 at 2 25 32 PM

### CODE REVIEW

File: frontend/src/components/pages/RegistrantExperience/index.tsx
Comment (e-wai) [inline frontend/src/components/pages/RegistrantExperience/index.tsx:79]: When do we need to skip more than one?

**Review by e-wai — CHANGES_REQUESTED** [review]
Some initial questions, lmk if I'm missing something

File: frontend/src/components/pages/RegistrantExperience/RegistrationFooter.tsx
Comment (bowenzhu1) [reply frontend/src/components/pages/RegistrantExperience/RegistrationFooter.tsx:52]: From talking with Andy + Dhruv: I think the intended behavior is that the button is always clickable, but will highlight field errors if there are any issues with the form.

File: frontend/src/components/pages/RegistrantExperience/index.tsx
Comment (bowenzhu1) [reply frontend/src/components/pages/RegistrantExperience/index.tsx:79]: There's no case where we skip more than one rn. But I also implemented like this to handle going back (-1 steps)

File: frontend/src/theme/index.ts
Comment (bowenzhu1) [reply frontend/src/theme/index.ts:31]: I think chakraui components at least let you define CSS based on breakpoints. Ex. Button in line 39 of RegistrationFooter.tsx `width={{ sm: "95vw", md: "45vw", lg: "auto" }}` . I don't think there's much responsiveness implemented anywhere else? but how it works in this case is if the screen width is larger than "sm"/320px, it uses width: "95vw", and so on for md and lg.

**Review by bowenzhu1 — COMMENTED** [review]
@e-wai  let me know if any changes should be made. I responded to your comments above. thanks!

File: frontend/src/theme/index.ts
Comment (dhmi19) [reply frontend/src/theme/index.ts:31]: These sizes don't seem to match the designs. Could you clarify with Andy to see what the correct sizes for each device should be?

**Review by dhmi19 — APPROVED** [review]
lgtm. I think the numeric approach for `stepNavigation` is nice and avoids code duplication. You could also do the same thing with Enums to define the actions and then the conditional logic would depend on the value of the enum. For example, have an enum with values: BACK_STEP, NEXT_STEP, PAYMENT_STEP, SESSION_SELECTION_STEP. that might make the code more readable. I am good to merge this pending Emma's approval though

File: frontend/src/components/pages/RegistrantExperience/index.tsx
Comment (e-wai) [reply frontend/src/components/pages/RegistrantExperience/index.tsx:79]: Hmm the default doesn't help much with clarity -- can we at least include a type?

**Review by e-wai — CHANGES_REQUESTED** [review]

**Comment by bowenzhu1** [comment]
oh I merged instead of rebasing, should I redo that or is it fine to squash and merge this

**Review by e-wai — APPROVED** [review]
Nope, we have squash and merge so good to go! Thanks for iterating 🙏

**Pull request merged**
